sharding/nodesSetup_test.go

```diff
@@ -971,212 +971,90 @@ func TestNodesSetup_NumberOfShardsShouldWork(t *testing.T) {
-		shardMinNodes:      2,
-		metaConsensusSize:  1,
-		metaMinNodes:       1,
-		numInitialNodes:    3,
-		genesisMaxShards:   1,
+		nrShards := ns.MinNumberOfShardNodes()
+		require.Equal(t, uint32(1), nrShards)
 	})
-	require.Nil(t, err)
-	ns.Hysteresis = 0.2
-	ns.Adaptivity = false
 
-	ns = createAndAssignNodes(ns, noOfInitialNodes)
+	t.Run("MinNumberOfMetaNodes", func(t *testing.T) {
+		t.Parallel()
 
-	shardConsensusGroupSize := ns.GetShardConsensusGroupSize()
-
-	require.Equal(t, uint32(2), shardConsensusGroupSize)
-}
-
-func TestNodesSetup_GetMetaConsensusGroupSize(t *testing.T) {
-	t.Parallel()
-
-	noOfInitialNodes := 3
-
-	ns, err := createTestNodesSetup(argsTestNodesSetup{
-		shardConsensusSize: 1,
-		shardMinNodes:      1,
-		metaConsensusSize:  2,
-		metaMinNodes:       2,
-		numInitialNodes:    3,
-		genesisMaxShards:   1,
+		nrShards := ns.MinNumberOfShardNodes()
+		require.Equal(t, uint32(1), nrShards)
 	})
-	require.Nil(t, err)
-	ns.Hysteresis = 0.2
-	ns.Adaptivity = false
 
-	ns = createAndAssignNodes(ns, noOfInitialNodes)
-
-	metaConsensusGroupSize := ns.GetMetaConsensusGroupSize()
-
-	require.Equal(t, uint32(2), metaConsensusGroupSize)
-}
+	t.Run("GetShardConsensusGroupSize", func(t *testing.T) {
+		t.Parallel()
 
-func TestNodesSetup_GetHysteresis(t *testing.T) {
-	t.Parallel()
-
-	ns, _ := createTestNodesSetup(argsTestNodesSetup{
-		shardConsensusSize: 1,
-		shardMinNodes:      1,
-		metaConsensusSize:  1,
-		metaMinNodes:       1,
-		numInitialNodes:    3,
-		genesisMaxShards:   1,
+		shardConsensusGroupSize := ns.GetShardConsensusGroupSize()
+		require.Equal(t, uint32(1), shardConsensusGroupSize)
 	})
-	require.NotNil(t, ns)
-	ns.Hysteresis = 0.5
-
-	hysteresis := ns.GetHysteresis()
 
-	require.Equal(t, float32(0.5), hysteresis)
-}
-
-func TestNodesSetup_GetAdaptivity(t *testing.T) {
-	t.Parallel()
+	t.Run("GetMetaConsensusGroupSize", func(t *testing.T) {
+		t.Parallel()
 
-	ns, _ := createTestNodesSetup(argsTestNodesSetup{
-		shardConsensusSize: 1,
-		shardMinNodes:      1,
-		metaConsensusSize:  1,
-		metaMinNodes:       1,
-		numInitialNodes:    3,
-		genesisMaxShards:   1,
+		metaConsensusGroupSize := ns.GetMetaConsensusGroupSize()
+		require.Equal(t, uint32(1), metaConsensusGroupSize)
 	})
 
-	require.NotNil(t, ns)
-	ns.Adaptivity = true
-	adaptivity := ns.GetAdaptivity()
-
-	require.True(t, adaptivity)
-}
-
-func TestNodesSetup_GetStartTime(t *testing.T) {
-	t.Parallel()
+	t.Run("GetHysteresis", func(t *testing.T) {
+		t.Parallel()
 
-	ns, _ := createTestNodesSetup(argsTestNodesSetup{
-		shardConsensusSize: 1,
-		shardMinNodes:      1,
-		metaConsensusSize:  1,
-		metaMinNodes:       1,
-		numInitialNodes:    3,
-		genesisMaxShards:   1,
+		ns.Hysteresis = 0.5
+		hysteresis := ns.GetHysteresis()
+		require.Equal(t, float32(0.5), hysteresis)
 	})
 
-	require.NotNil(t, ns)
+	t.Run("GetAdaptivity", func(t *testing.T) {
+		t.Parallel()
 
-	ns.StartTime = 2
-	startTime := ns.GetStartTime()
+		ns.Adaptivity = true
```

Comment:
because these tests run in paralel, they may randomly fail due to this change here and the one on L1028.. I suggest removing all t.Runs and simply call the getters one by one

Reply:
updated, removed t.Run